File: agent_misc/mcps/codeql-development-mcp-server/server/ql/ruby/tools/test/PrintCFG/Example1.rb
# Example1 test case for PrintCFG.ql

# COMPLIANT - Basic control flow for CFG testing
def simple_method(x)
  if x > 0
    puts "Positive"
  else
    puts "Non-positive"
  end

  (0..2).each do |i|
    puts i
  end
end

# NON_COMPLIANT - Method with complex control flow
def complex_method(value)
  return -1 if value < 0

  while value > 10
    value = value / 2
  end

  case value
  when 0
    0
  when 1
    1
  else
    value * 2
  end
end

File: agent_misc/mcps/codeql-development-mcp-server/server/ql/javascript/tools/test/PrintCFG/Example1.js
// Example1 test case for PrintCFG.ql

// COMPLIANT - Basic control flow for CFG testing
function simpleMethod(x) {
    if (x > 0) {
        console.log("Positive");
    } else {
        console.log("Non-positive");
    }

    for (let i = 0; i < 3; i++) {
        console.log(i);
    }
}

// NON_COMPLIANT - Function with complex control flow
function complexMethod(value) {
    if (value < 0) {
        return -1;
    }

    while (value > 10) {
        value = value / 2;
    }

    switch (value) {
        case 0:
            return 0;
        case 1:
            return 1;
        default:
            return value * 2;
    }
}
